llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -19089,7 +19089,8 @@ static bool shouldCombineToPostInc(SDNode *N, SDValue Ptr, SDNode *PtrUse,
                                    IsMasked, OtherPtr, TLI)) {
         SmallVector<const SDNode *, 2> Worklist;
         Worklist.push_back(Use);
-        if (SDNode::hasPredecessorHelper(N, Visited, Worklist))
+        constexpr unsigned int MaxSteps = 8192;
```

Comment:
Big number to repeat multiple times. Move these to a common constant?

Reply:
I changed it to be a hidden cl option in case someone would like to play with it later, and defaulted to 8192. WDYT?